### DISCUSSION TITLE
git@github.com: Permission denied (publickey)

### DISCUSSION THREAD

**Discussion by gopalepic** [community/community#79366] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

i am getting the error
{
 git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
}
 while i am trying to push my code to github from terminal

**Comment by sh0rtener** [comment]
The error tells you that you are either not authenticated or you do not have access to the repository. You can tell us how you get the push code. In any case, you need to authenticate first, if you haven't already done so :)

Resources that can help:
https://docs.github.com/ru/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens
https://docs.github.com/ru/authentication/troubleshooting-ssh/error-permission-denied-publickey
